### PR TITLE
An Introduction to Testing for Accessibility

### PR DESCRIPTION
**Pull request by hxmattobee**
### Blog title
An Introduction to Testing for Accessibility

### Blog description
A relatively high-level introduction to accessibility. How can testers and other members of the team learn about accessibility? How can we design inclusive experiences and test for accessibility barriers?

I have also updated my job title in two places at the same time.

#### Author
- [x] I have checked for grammatical errors
- [x] I have checked for spelling errors
- [x] All information (technical or otherwise) is correct to my knowledge

#### Editor 1
- [x] I have checked for grammatical errors
- [x] I have checked for spelling errors
- [x] All information (technical or otherwise) is correct to my knowledge

#### Editor 2
- [x] I have checked for grammatical errors
- [x] I have checked for spelling errors
- [x] All information (technical or otherwise) is correct to my knowledge

### CODE REVIEW

**Comment by alannamooney** [comment]
Taking 1st editor slot

File: _posts/2015-11-24-testing-for-accessibility.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+The main international standard for the accessibility of Web content is [<cite><abbr title="Web Content Accessibility Guidelines">WCAG</abbr></cite> 2.0](http://www.w3.org/TR/WCAG20/), also known as <cite>ISO/IEC 40500:2012</cite>, published by <abbr title="World Wide Web Consortium">W3C</abbr>. While aimed primarily at Web content, [<cite><abbr title="Web Content Accessibility Guidelines">WCAG</abbr></cite> can also be applied to native mobile applications](http://www.w3.org/TR/mobile-accessibility-mapping/). It's against these guidelines that most developers and service providers evaluate their products.
```
Comment (alannamooney) [inline _posts/2015-11-24-testing-for-accessibility.md]: has the second <cite> been closed in the wrong place? - 'can also be applied to native mobile applications currently looks like part of the link, should it be?

File: _posts/2015-11-24-testing-for-accessibility.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+Accessibility is however much more than a box ticking exercise.
```
Comment (alannamooney) [inline _posts/2015-11-24-testing-for-accessibility.md]: however should have a comma either side.

File: _posts/2015-11-24-testing-for-accessibility.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+According to [AbilityNet](https://www.abilitynet.org.uk/advice-information/enation/travel-and-tourism-summer-july-2014) (2014) there are 12 million disabled people and 10 million people over 65 in the UK, with an estimated spending power of over £100<abbr title="billion">bn</abbr>. If you ignore accessibility, you ignore a huge group of customers and a lucrative market.
+
+Beyond the legal and commercial arguments, making digital products and services accessibile to everyone is simply _the right thing to do_. Everyone deserves to be included.
```
Comment (alannamooney) [inline _posts/2015-11-24-testing-for-accessibility.md]: accessible has an extra i in it

File: _posts/2015-11-24-testing-for-accessibility.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+## Recognise the diversity
+
+When you think about accessibility, you probably imagine a blind person using a screen reader (a piece of software that speaks everything that appears on screen). That's understandable because visual impairment affects lots of people in one way or another and screen readers are one of the most ubiquitous forms of assistive technology, so it's easy for us to empathise. The big picture however is far more complex and diverse. Just a few examples include:
```
Comment (alannamooney) [inline _posts/2015-11-24-testing-for-accessibility.md]: 'however' in the last sentence should have a comma either side

File: _posts/2015-11-24-testing-for-accessibility.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+* Someone with colourblindness who has difficulty distinguishing between colours such as red and green;
+* A deaf person who relies on captions or transcripts for audio content;
+* Someone who has reduced dexterity and is unable to use a mouse;
+* Someone with photosensitive epilepsy who experiences seizures when exposed to certain flashing patterns;
+* A person with a learning disabilitiy like dyslexia who has trouble reading;
```
Comment (alannamooney) [inline _posts/2015-11-24-testing-for-accessibility.md]: disability has one too many i's in it

File: _posts/2015-11-24-testing-for-accessibility.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+There are lots more examples in [An Alphabet of Accessibility Issues](https://the-pastry-box-project.net/anne-gibson/2014-july-31) and another great resource is the <abbr title="Web Accessibility Initiative">WAI</abbr>’s [How People with Disablities Use the Web](http://www.w3.org/WAI/intro/people-use-web/Overview.html) which provides more information about the problems that these people experience.
+
+If you have the opportunity, the best way to learn about accessibility is of course to spend time with the people in these groups and see for yourself the problems that they face.
```
Comment (alannamooney) [inline _posts/2015-11-24-testing-for-accessibility.md]: of course should have commas either side

File: _posts/2015-11-24-testing-for-accessibility.md [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+
+## Use tools but don't rely exclusively on automation
+
+There are a number of services and applications that aim to test websites for accessibility. Give them a URL and they'll produce a report that shows all of the passes, failures and warnings (according to best practices or formal standards such as <abbr title="Web Content Accessibility Guidelines">WCAG</abbr>).
+
+The problem with these automated systems is that they can only check for the most rudimentary accessibility problems. Like any other automated checking tool, they don't understand context. For example, tools can find an `img` element in an HTML document and check whether or not it has a text alternative in an `alt` attribute, but they can't decide whether or not that text alternative is appropriate or well written.
+
+One such tool is [Tennon](http://www.tenon.io/documentation/what-tenon-tests.php) and its developers make a point of acknowledging its limitations:
```
Comment (alannamooney) [inline _posts/2015-11-24-testing-for-accessibility.md]: the link is spelt 'Tennon' but the URL has 'Tenon' - which one is right?

File: _posts/2015-11-24-testing-for-accessibility.md [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+
+## Expert testing or user testing?
+
+Whenever possible, aim to involve real users in accessibility testing. This is no different from testing usability more generally. While a tester or developer can check an application against guidelines and best practices, they will always be biased by their insider knowledge. Also, while they might be technical experts, testers and developers are not experts at using assistive technology in the same way that real users are.
+
+There are however lots of things that a developer or expert tester can and should be checking for, even before users are involved. For example, the following quick and easy checks can be performed on any website, perhaps before a more in-depth audit takes place:
```
Comment (alannamooney) [inline _posts/2015-11-24-testing-for-accessibility.md]: however should have commas either side

**Comment by alannamooney** [comment]
I've picked a few grammatical errors and spelling mistakes, happy to +1 once they've been addressed :-)

File: _posts/2015-11-24-testing-for-accessibility.md (hunk 1 above)
Comment (hxmattobee) [inline _posts/2015-11-24-testing-for-accessibility.md]: Yep, that was intentional.

**Comment by hxmattobee** [comment]
Fixed, @alannamooney. Thank you for that.

**Comment by dancork** [comment]
2nd editor

File: _posts/2015-11-24-testing-for-accessibility.md [hunk 9]
```diff
@@ -0,0 +1,81 @@
+---
```
Comment (dancork) [inline _posts/2015-11-24-testing-for-accessibility.md]: Can we update the date in the filename to today

File: _posts/2015-11-24-testing-for-accessibility.md [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+layout: post
+title: An Introduction to Testing for Accessibility
+categories: Testing
+author: matt_obee
+tags: testing accessibility
+comments: true
+excerpt: "How can testers and other members of the team learn about accessibility? How can we design inclusive experiences and test for accessibility barriers?"
+---
+
+When I gave a talk on <cite>Checking & Exploring</cite> at [The London Web meetup in October](http://www.meetup.com/londonweb/events/214733702/), a number of people asked questions or spoke to me in the bar afterwards about accessibility (for older people and those with disabilities). I only had time to touch on it very briefly in my presentation, but that doesn't mean it isn't important. Quite the opposite, it deserves to be covered in detail in its own right. Some people asked me how testers should be trained in accessibility and others asked what sort of testing we need to perform in order to find accessibility issues.
```
Comment (dancork) [inline _posts/2015-11-24-testing-for-accessibility.md]: A link to a video of the talk would be better - if you have one

File: _posts/2015-11-24-testing-for-accessibility.md [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+
+## Accessibility is part of design
+
+Firstly, an important disclaimer: Accessibility should be included as part of the design process; it's not just something that can be checked and bolted on at the end of the project. This means that responsibility for accessibility doesn't just lie with testers but with designers and developers too. Designers have to think about things like navigation methods, colour contrast and layout, while developers have to write code that exposes the right information to assistive technology (e.g. screen readers) in the right way. As [Michael Larsen](https://twitter.com/mkltesthead) said in his [presentation at Agile Testing Days 2015](http://www.agiletestingdays.com/session/making-a-web-for-everyone/), inclusive design is about making things accessible from the start with the minimum amount of retrospective change.
```
Comment (dancork) [inline _posts/2015-11-24-testing-for-accessibility.md]: If possible, a link to video of talk would be better if possible

File: _posts/2015-11-24-testing-for-accessibility.md (hunk 1 above)
Comment (dancork) [inline _posts/2015-11-24-testing-for-accessibility.md]: Should `Web content` be capitalised when not talking about the guidelines?

eg `standard for the accessibility of Web content`

Comment (dancork) [inline _posts/2015-11-24-testing-for-accessibility.md]: ...also `While aimed primarily at Web content`

File: _posts/2015-11-24-testing-for-accessibility.md [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+
+The main international standard for the accessibility of Web content is [<cite><abbr title="Web Content Accessibility Guidelines">WCAG</abbr></cite> 2.0](http://www.w3.org/TR/WCAG20/), also known as <cite>ISO/IEC 40500:2012</cite>, published by <abbr title="World Wide Web Consortium">W3C</abbr>. While aimed primarily at Web content, [<cite><abbr title="Web Content Accessibility Guidelines">WCAG</abbr></cite> can also be applied to native mobile applications](http://www.w3.org/TR/mobile-accessibility-mapping/). It's against these guidelines that most developers and service providers evaluate their products.
+
+Accessibility is, however, much more than a box ticking exercise.
```
Comment (dancork) [inline _posts/2015-11-24-testing-for-accessibility.md]: I would say `However, accessibility is much more than a box ticking exercise.` or just ditch the however.

File: _posts/2015-11-24-testing-for-accessibility.md [hunk 13]
```diff
@@ -0,0 +1,81 @@
...
+
+* Someone with colourblindness who has difficulty distinguishing between colours such as red and green;
+* A deaf person who relies on captions or transcripts for audio content;
+* Someone who has reduced dexterity and is unable to use a mouse;
+* Someone with photosensitive epilepsy who experiences seizures when exposed to certain flashing patterns;
+* A person with a learning disability like dyslexia who has trouble reading;
+
+There are lots more examples in [An Alphabet of Accessibility Issues](https://the-pastry-box-project.net/anne-gibson/2014-july-31) and another great resource is the <abbr title="Web Accessibility Initiative">WAI</abbr>’s [How People with Disablities Use the Web](http://www.w3.org/WAI/intro/people-use-web/Overview.html) which provides more information about the problems that these people experience.
```
Comment (dancork) [inline _posts/2015-11-24-testing-for-accessibility.md]: Not a big fan of the term `these people`

**Comment by dancork** [comment]
Great post :)

Instead of abbr I think you can use this http://kramdown.gettalong.org/syntax.html#abbreviations

File: _posts/2015-11-24-testing-for-accessibility.md (hunk 10 above)
Comment (hxmattobee) [inline _posts/2015-11-24-testing-for-accessibility.md]: I don't @dancork, sadly. It hasn't been published.

File: _posts/2015-11-24-testing-for-accessibility.md (hunk 11 above)
Comment (hxmattobee) [inline _posts/2015-11-24-testing-for-accessibility.md]: There isn't a video yet, @dancork. Or at least it isn't publicly available.

**Comment by hxmattobee** [comment]
All sorted, thanks @dancork.

**Comment by dancork** [comment]
Awesome :+1:

**Pull request merged**